### PR TITLE
Add Permissions Policy Integration and initial createContext() hooks

### LINKED ISSUE #145: Make WebNN API a policy-controlled feature with default allowlist 'self'
[PING review feedback]( https://github.com/webmachinelearning/webnn/issues/119#issuecomment-772565085):

>Is the API restricted to first-party contexts? Or do third-party frames have access? (The answer to 2.13 of the Self-Review: Security and Privacy Questionnaire (above) suggests they do, and that you are exploring the potential of a policy-controlled feature approach.) Is there any reason not to simply restrict to first party context? (i.e. what are the likely use cases you envision that would require third-party frames to have access to the API?)

Discussed on [WebML CG Teleconference – 18 February 2021](https://www.w3.org/2021/02/18-webmachinelearning-minutes.html#t09) with an initial agreement to make WebNN API a [policy controlled feature](https://w3c.github.io/webappsec-permissions-policy/#policy-controlled-feature).

We have additional knobs at our disposal with the [default allowlist](https://w3c.github.io/webappsec-permissions-policy/#default-allowlist) with two possible values:

>`*`
>The feature is allowed in documents in top-level browsing contexts by default, and when allowed, is allowed by default to documents in child browsing contexts.
>`self`
>The feature is allowed in documents in top-level browsing contexts by default, and when allowed, is allowed by default to same-origin domain documents in child browsing contexts, but is disallowed by default in cross-origin documents in child browsing contexts

Per PING feedback I infer we should set the default allowlist to `['self']` that allow same-origin domain iframe elements have access to this API, while disallow by default cross-origin iframe access to this API.

@rafaelcintron @sandandsnow

### PR DESCRIPTION
**Pull request by anssiko**
Fix #145

In addition to Permissions Policy Integration (per [integration best practices](https://github.com/w3c/webappsec-permissions-policy/blob/main/integration.md)), this also adds initial steps for `createContext()` to allow integration of the "allowed to use" check.

@wchao1115 you wanted to clarify the context creation prose so if this looks good to you you can base your further work on top of this and augment the steps with more context creation details.

***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/pull/159.html" title="Last updated on Apr 7, 2021, 3:22 PM UTC (ce01a73)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/159/0f913d6...ce01a73.html" title="Last updated on Apr 7, 2021, 3:22 PM UTC (ce01a73)">Diff</a>

### CODE REVIEW

**Review by wchao1115 — APPROVED** [review]

**Comment by anssiko** [comment]
@wchao1115 feel free to merge this PR if it helps unblock your other work. You can also stack your PR on top of this, but I believe in that case we'd need to start merging "from the top" i.e. the latest opened PR to avoid merge conflicts.

I'll let you own this workflow with @huningxin so feel free to pick a workflow that works for both of you editors :-)

**Comment by anssiko** [comment]
@wchao1115 @huningxin per discussion [today's call](https://www.w3.org/2021/04/15-webmachinelearning-minutes.html#t02), feel free to merge this. I'll let you do the merge to coordinate with other open PRs.

**Review by huningxin — APPROVED** [review]
LGTM, thanks @anssiko !

**Pull request merged**
